applications/FluidDynamicsApplication/tests/cpp_tests/test_adjoint_schemes.cpp

```diff
@@ -0,0 +1,807 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+//
+
+// System includes
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/cfd_variables.h"
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "spaces/ublas_space.h"
+#include "testing/testing.h"
+#include "utilities/normal_calculation_utils.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "custom_response_functions/velocity_pressure_norm_square_response_function.h"
+#include "custom_strategies/schemes/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h"
+#include "custom_strategies/schemes/residualbased_simple_steady_scheme.h"
+#include "custom_strategies/schemes/simple_steady_adjoint_scheme.h"
+#include "custom_strategies/schemes/velocity_bossak_adjoint_scheme.h"
+#include "custom_strategies/schemes/simple_steady_sensitivity_builder_scheme.h"
+#include "custom_strategies/schemes/velocity_bossak_sensitivity_builder_scheme.h"
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+array_1d<double, 3> GetArray3DVector(
+    const std::vector<double>& rInput)
+{
+    array_1d<double, 3> result;
+    result[0] = rInput[0];
+    result[1] = rInput[1];
+    result[2] = rInput[2];
+
+    return result;
+}
```

Comment:
I think you can avoid this with the new `array_1d` constructor (it accepts an initializer list).

Reply:
done :)